### PR TITLE
Create CONTRIBUTING.md

### PR DESCRIPTION
**Pull request by devendran-m**
Creating CONTRIBUTING document for Casper Ecosystem

### CODE REVIEW

File: profile/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+
+**Step 5:** Casper Team will review the Pull Request and Accept if the changes are in line with Casper’s expected code quality, test coverage and security standards
+
+- **Step 5.1:** Git Rules
+
+    We use `gitchangelog` for all of our repos for change logs. For that, we need to follow the following convention when writing commit messages. Your pull request may not be merged if you are not following this convention.
+
+- **Step 5.2:** Commit Message Convention
+
+    The following are suggestions to what might be useful to think about adding in your commit messages. You might want to separate roughly your commits into big sections:
```
Comment (ACStoneCL) [inline profile/CONTRIBUTING.md:100]:
```suggestion
    The following suggestions might be useful to add in your commit messages. You might want to separate your commits into rough sections:
```

File: profile/CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+    - by purpose (for example: new, fix, change ...)
+    - by the entity (for example: doc, sdk, code ...)
+    - by the audience (for example: dev, tester, users ...)
+  
+    Additionally, you could want to tag some commits:
```
Comment (ACStoneCL) [inline profile/CONTRIBUTING.md:105]:
```suggestion
    Additionally, you might want to tag some commits:
```

File: profile/CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    - as `minor` commits that shouldn’t get output to your changelog (cosmetic changes, small typo in comments...)
+    - as `refactor` if you don’t really have any significant feature changes
```
Comment (ACStoneCL) [inline profile/CONTRIBUTING.md:107]:
```suggestion
    - as `refactor` if you don’t have any significant feature changes
```

File: profile/CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,168 @@
...
+        Thus, this should not also be part of the changelog displayed to a final user for instance, but might be of some interest if you have a developer changelog.
```
Comment (ACStoneCL) [inline profile/CONTRIBUTING.md:108]:
```suggestion
        Thus, this should not also be part of the changelog displayed to a final user, but might be of some interest if you have a developer changelog.
```

File: profile/CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,168 @@
...
+Try writing your commit message by targeting user functionality as often as you can. You can also select existing labels/tags available on the repos to categorize your commit message.
+
+**Casper Pull Request Workflow:**
+
+Please refer to the Casper Pull Request Workflow below;
+
+![Casper PR Workflow](https://github.com/casper-ecosystem/.github/new/main/CASPERPRFLOW.jpeg)
+
+- **Step 5.3:** Best Practices
+  - Code in the Casper project should meet the style guidelines, with sufficient test cases, descriptive commit messages, evidence that the contribution does not break any compatibility commitments or cause adverse feature interactions, and evidence of high-quality peer review.
```
Comment (ACStoneCL) [inline profile/CONTRIBUTING.md:121]:
```suggestion
  - Code in the Casper project should meet the style guidelines with sufficient test cases, descriptive commit messages, evidence that the contribution does not break any compatibility commitments or cause adverse feature interactions and evidence of high-quality peer review.
```

File: profile/CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,168 @@
...
+  - When you notice those dependencies, put the fix into a commit of its own, then check out a new branch, and cherry-pick it.
+  - Read the PR template prudently and make sure to follow all the instructions.
+  - Please pay attention to the maintainer's feedback, since it is a necessary step to keep up with the standards Casper attains.
+  - If the feature requires ongoing maintenance (e.g. support for a particular brand of database), we may ask you to accept responsibility for maintaining this feature.
+
+
+
+### 5. Testing
+
+If you are interested in testing our documentation, guides and developer tools, head over to the following pages.
```
Comment (ACStoneCL) [inline profile/CONTRIBUTING.md:132]:
```suggestion
If you are interested in testing our documentation, guides and developer tools, head over to the following pages:
```

File: profile/CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,168 @@
...
+
+- [Why Build on Casper](https://docs.casperlabs.io/dapp-dev-guide/why-build-on-casper/)
+- [Getting Started with Rust](https://docs.casperlabs.io/dapp-dev-guide/writing-contracts/getting-started/)
+- [Setting up a Local Network with NCTL](https://docs.casperlabs.io/dapp-dev-guide/building-dapps/setup-nctl/)
+- [Testing Smart Contracts with NCTL](https://docs.casperlabs.io/dapp-dev-guide/building-dapps/nctl-test/)
+
+We would love to know your insights/feedback from your testing, and you can let us know by submitting issues in the respective repositories on GitHub.
```
Comment (ACStoneCL) [inline profile/CONTRIBUTING.md:139]:
```suggestion
We would love to know your insights/feedback from your testing. You can let us know by submitting issues in the respective repositories on GitHub.
```

File: profile/CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,168 @@
...
+
+### 6. Contributing New Projects to Casper Ecosystem
+
+This section is specifically created to explain the mandatory requirements that the Community Developers/Teams are required to follow when contributing new projects to Casper Ecosystem.
```
Comment (ACStoneCL) [inline profile/CONTRIBUTING.md:143]:
```suggestion
This section is specifically created to explain the mandatory requirements that the Community Developers/Teams are required to follow when contributing new projects to the Casper Ecosystem.
```

File: profile/CONTRIBUTING.md [hunk 9]
```diff
@@ -0,0 +1,168 @@
...
+- Code of conduct (CODE_OF_CONDUCT.md)
+- Contribution policy (CONTRIBUTING.md)
+- Security policy (SECURITY.md)
+
+**Nice to have**
+- Issue templates
+- PR template
+- Repository admins accept content reports (enabled from the admin settings of the repo)
+
+## Contributing Style guide
```
Comment (ACStoneCL) [inline profile/CONTRIBUTING.md:164]:
```suggestion
## Contributing Style Guide
```

**Comment by devendran-m** [comment]
Thank you, Adam (@ACStoneCL) for reviewing the content. I've incorporated the comments.

**Pull request merged**